frontend/src/actionCreators/partiesActionCreator.js

```diff
@@ -222,7 +198,73 @@ export const fetchPartyRelationships = (mineId) => (dispatch) => {
         duration: 10,
       });
       dispatch(error(reducerTypes.FETCH_PARTY_RELATIONSHIPS));
-      dispatch(hideLoading("modal"));
+      dispatch(hideLoading());
+    });
+};
+
+export const fetchPartyRelationshipsByMineId = (mineId) => (dispatch) => {
+  dispatch(request(reducerTypes.FETCH_PARTY_RELATIONSHIPS));
+  dispatch(showLoading());
+  return axios
+    .get(
+      `${ENVIRONMENT.apiUrl + API.PARTY_RELATIONSHIP}?mine_guid=${mineId}`,
+      createRequestHeader()
+    )
+    .then((response) => {
+      dispatch(success(reducerTypes.FETCH_PARTY_RELATIONSHIPS));
+      dispatch(partyActions.storePartyRelationships(response.data));
+      dispatch(hideLoading());
+    })
+    .catch((err) => {
+      notification.error({
+        message: err.response ? err.response.data.error.message : String.ERROR,
+        duration: 10,
+      });
+      dispatch(error(reducerTypes.FETCH_PARTY_RELATIONSHIPS));
+      dispatch(hideLoading());
+    });
+};
+
+export const fetchPartyRelationshipsByPartyId = (partyId) => (dispatch) => {
+  dispatch(request(reducerTypes.FETCH_PARTY_RELATIONSHIPS));
+  dispatch(showLoading());
+  return axios
+    .get(
+      `${ENVIRONMENT.apiUrl + API.PARTY_RELATIONSHIP}?party_guid=${partyId}`,
+      createRequestHeader()
+    )
+    .then((response) => {
+      dispatch(success(reducerTypes.FETCH_PARTY_RELATIONSHIPS));
+      dispatch(partyActions.storePartyRelationships(response.data));
+      dispatch(hideLoading());
+    })
+    .catch((err) => {
+      notification.error({
+        message: err.response ? err.response.data.error.message : String.ERROR,
+        duration: 10,
+      });
+      dispatch(error(reducerTypes.FETCH_PARTY_RELATIONSHIPS));
+      dispatch(hideLoading());
+    });
+};
+
+export const fetchPartyRelationshipsByTypeCode = (typeCode) => (dispatch) => {
+  dispatch(request(reducerTypes.FETCH_PARTY_RELATIONSHIPS));
+  dispatch(showLoading());
+  return axios
+    .get(`${ENVIRONMENT.apiUrl + API.PARTY_RELATIONSHIP}?types=${typeCode}`, createRequestHeader())
+    .then((response) => {
+      dispatch(success(reducerTypes.FETCH_PARTY_RELATIONSHIPS));
+      dispatch(partyActions.storePartyRelationships(response.data));
+      dispatch(hideLoading());
+    })
+    .catch((err) => {
+      notification.error({
+        message: err.response ? err.response.data.error.message : String.ERROR,
+        duration: 10,
+      });
+      dispatch(error(reducerTypes.FETCH_PARTY_RELATIONSHIPS));
+      dispatch(hideLoading());
     });
```

Comment:
These three fetchParty... methods could probably be combined or call one inner method to avoid duplication.

Reply:
Agree, that was the intent but I didn't refactor the other uses. Done.